src/main/java/org/broadinstitute/hellbender/engine/ReadShard.java

```diff
@@ -0,0 +1,182 @@
+     * Reads in this shard will be filtered using this filter before being returned
+     *
+     * @param filter filter to use (may be null)
+     */
+    public void setReadFilter( final ReadFilter filter ) {
+        this.readFilter = filter;
+    }
+
+    /**
+     * @return Contig this shard belongs to
+     */
+    @Override
+    public String getContig() {
+        return interval.getContig();
+    }
+
+    /**
+     * @return Start position of this shard
+     */
+    @Override
+    public int getStart() {
+        return interval.getStart();
+    }
+
+    /**
+     * @return End position of this shard
+     */
+    @Override
+    public int getEnd() {
+        return interval.getEnd();
+    }
+
+    /**
+     * @return the interval this shard spans
+     */
+    public SimpleInterval getInterval() {
+        return interval;
+    }
+
+    /**
+     * @return the interval this shard spans, potentially with additional padding on each side
+     */
+    public SimpleInterval getPaddedInterval() {
+        return paddedInterval;
+    }
+
+    /**
+     * @return an iterator over reads in this shard, as filtered using the configured read filter;
+     *         reads are lazily loaded rather than pre-loaded
+     */
+    @Override
+    public Iterator<GATKRead> iterator() {
+        final Iterator<GATKRead> readsIterator = readsSource.query(paddedInterval);
+
+        return readFilter != null ? new ReadFilteringIterator(readsIterator, readFilter) : readsIterator;
+    }
+
+    /**
+     * @return a List containing all reads in this shard, pre-loaded, and filtered using the configured read filter
+     *
+     * note: call {@link #iterator} instead to avoid pre-loading all reads at once
+     */
+    public List<GATKRead> loadAllReads() {
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator(), Spliterator.ORDERED), false).collect(Collectors.toList());
+    }
+
+    /**
+     * @param loc Locatable to test
+     * @return true if loc is contained within this shard's interval, otherwise false
+     */
+    public boolean isContainedWithinShard( final Locatable loc ) {
+        return interval.contains(loc);
+    }
+
+    /**
+     * @param loc Locatable to test
+     * @return true if loc starts within this shard's interval, otherwise false
+     */
+    public boolean startsWithinShard( final Locatable loc ) {
+        return interval.contains(new SimpleInterval(loc.getContig(), loc.getStart(), loc.getStart()));
+    }
+
+    /**
+     * @return number of bases of padding to the left of our interval
+     */
+    public int numLeftPaddingBases() {
+        return interval.getStart() - paddedInterval.getStart();
+    }
+
+    /**
+     * @return number of bases of padding to the right of our interval
+     */
+    public int numRightPaddingBases() {
+        return paddedInterval.getEnd() - interval.getEnd();
+    }
+
+    /**
+     * Shard an interval into ReadShards using the shardSize and shardPadding arguments
```

Comment:
i'm not a literary man but `Shard an interval into ReadShards using the shardSize and shardPadding arguments` sounds a bit too shard-full for me.